pkg/github/github_client.go

```diff
@@ -80,32 +83,69 @@ func newGithubClientByApp(appId int64, privateKeyPem []byte, owner string) (*Git
 	return NewGithubClient(token.GetToken()), nil
 }
 
-func (c *GithubClient) GetReference(owner, repository, branch string) (*github.Reference, error) {
+func (c *GithubClient) isReferenceExist(owner, repository, branch string) (bool, error) {
+	_, resp, err := c.client.Git.GetRef(c.ctx, owner, repository, "refs/heads/"+branch)
+	if err == nil {
+		return true, nil
+	}
+
+	if resp.StatusCode == 404 {
+		return false, nil
+	}
+	return false, err
```

Comment:
while I did somewhat, without backtracking all the error propagation @mmorhun, so perhaps what I'm about to mention is "handled", but from what I saw, I *think* retry will occur if a non-404 error occurs (of course correct me if I'm wrong).

if so, there are some other errors besides 404, like permission denied, which are permanent failures which most likely will not go away anytime soon, and some like throttling which might go away relatively soon, and warrant a retry

and hence it could make sense to broaden our http resp code analysis somewhat 

http internal errors, client/server timeouts, connection reset, unexpected eof, etc.

Reply:
You are right, @gabemontero , the errors must be handled better. We already have a separate issue about that.
In this PR, I am fixing logic of PR creation. Errors handling is untouched.

Reply:
I'm good tracking with the tracking story for handling k8s client non 200 http rc's separately / holistically across the broader component / appstudio even @mmorhun